File: payment-demo-ui/src/types/payment.ts
export type PaymentStatus = "Created" | "Authorised" | "Captured" | "Refunded";

export type CreatePaymentRequest = {
  amount: number;
  currency: string;
  merchantReference: string;
  cardLast4: string;
};

export type PaymentSummary = {
  id: string;
  amount: number;
  currency: string;
  merchantReference: string;
  cardLast4: string;
  createdAt: string;
  status: PaymentStatus;
};

export type CreatePaymentResult = {
  payment: PaymentSummary;
  idempotentReplay: boolean;
};

export type PaymentHistoryEntry = {
  id: number;
  paymentId: string;
  at: string;
  status: PaymentStatus;
  note: string;
};

File: PaymentGatewayMvpApp/Domain/Payment.cs
namespace PaymentGatewayMvpApp.Domain;

public sealed class Payment
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public decimal Amount { get; set; }
    public string Currency { get; set; } = string.Empty;
    public string MerchantReference { get; set; } = string.Empty;
    public string CardLast4 { get; set; } = string.Empty;
    public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;
    public Status Status { get; set; } = Status.Created;
    public List<History> History { get; set; } = new();
}
